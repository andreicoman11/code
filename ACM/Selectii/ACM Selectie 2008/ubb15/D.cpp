#include <iostream>
#include <fstream>
#include <vector>
#include <string>
#include <sstream>
#include <map>
#include <list>
#include <cmath>
#include <algorithm>
#include <set>
#include <iomanip>
using namespace std;

ifstream fin ("D.in");
//ofstream fout ("D.out");

int d;

int main()
{
	int c[400], a[400];
	fin >> d;
	while(d)
	{
		double p;
		memset(c, 0, sizeof(c));
		memset(a, 0, sizeof(a));
		fin >> p;
		int i = 0;
		c[0] = 100000;
		a[0] = (int)( (double)c[0]/p*0.97 );
		//cout << i << " " << c[i] << " " << a[i] << "\n";
		for(i=1; i<d; i++)
		{
			fin >> p;
			int ci = max( c[i-1], (int)( (double)a[i-1]*p*0.97 ) );
			int ai = max( a[i-1], (int)( (((double)c[i-1])/p)*0.97 ) );
			c[i] = ci;
			a[i] = ai;
			//cout << i << " " << c[i] << " " << a[i] << " " << "\n";
		}

		double rez = c[d-1];
		rez /= 100;
		cout.setf(ios::fixed);
		cout << setprecision(2) << rez << "\n";

		fin >> d;
	}

	return 0;
}