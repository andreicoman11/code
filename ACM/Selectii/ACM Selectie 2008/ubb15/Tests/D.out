2395.09
2736.89
2300.04
2627.05
2672.67
2265.24
2641.88
2329.42
2979.65
2154.07
2302.12
2351.92
2068.06
2836.75
2598.80
2632.68
2770.22
2290.59
2450.80
2105.60
2377.42
2453.06
2983.78
2378.12
2285.70
2368.43
2258.89
2117.61
2668.49
3340.57
